### ISSUE TITLE
[REX Google Cloud Deep Learning VM: Le ML dans le Cloud de façon simple ?] Florient Chouteau

### ISSUE THREAD

**Issue by fchouteau** [TlseDataScience/meetup-cfp#24]
## Le speaker
Florient Chouteau, Ingénieur Deep Learning à Airbus Defence and Space

* Twitter handle: @foxchouteau

## Sujet de votre conférence
Retour d'Expérience Google Cloud Deep Learning VM: Le ML dans le Cloud de façon simple ?

## Description de votre conférence

- Vous avez toujours rêvé de commencer le deep learning, ou une compétition Kaggle, mais n'avez pas de GPU chez vous ? 
- Vous avez un beau script et frôlez le top 100 d'une compétition et il ne vous manque qu'un peu de puissance de calcul pour passer à l'échelle ? 
- Vous donnez un cours de Deep Learning dans une école qui ne permet pas le proxy SSH, à des élèves dont les PC pourront à peine entraîner MNIST dans la durée du cours, mais vous disposez de quelques crédits GCP ?

Généralement, le cloud est une réponse envisageable à la majorité des problèmes ci-dessus. Mais aller instancier une machine virtuelle dans un des datacenters de votre provider favoris (Azure, Amazon, Google, et les multiples autres offrant des interfaces pour le ML en particulier) est parfois un casse tête:
- Ca peut très rapidement couter un certain prix
- Il faut passer du temps à tout installer (et donc ça monte le prix)
- Tout installer correctement n'est pas trivial, surtout si l'on débute

Comment maximiser l'intérêt tout en minimisant le cout ? 

Ce quickie / journal club session propose un retour d'expérience Google Cloud Platform nommé ["Deep Learning VM"](https://cloud.google.com/deep-learning-vm/) (ou aussi ["Google Cloud ML Notebook"](https://cloud.google.com/ml-engine/docs/notebooks/) ).

Ces machines sont des instances **pré-installées très correctement**, avec beaucoup de "petits bonus" orientés autour de la gestion et de l'éxécution de notebooks jupyters. 
Notamment:
- Un large choix de machine (tensorflow, pytorch, fastai, xgboost)
- Un serveur jupyter lancé automatiquement et accessible hors tunnel ssh (très utilise pour l'enseignement)
- Des petits add-ons github/jupyter diff
- La possibilité d'utiliser les machines avec Google Collab
- La possibilité d'éxecuter un notebook jupyter sur une machine qui se détruit automatiquement à la fin (utile pour la gestion des couts) grâce à `Papermill`.

L'idée du quickie est de montrer en live comment ça se passe en pratique (avec un cas d'utilisation extrêmement simple)... Et bien sur d'aborder l'aspect financier, ainsi que les points faibles du produit.

## Informations diverses
* Thématique, Labels : Tech Demo, Google Cloud Platform, Jupyter Notebook, "Hardware" for Machine Learning
* Niveau de difficulté (débutant|avancé|confirmé) : débutant
* Durée (quickie 15-30min, conférence >1h) :  Présentation journal club + quickie (20/30 minutes)
* Format (slides, live-coding, les deux): Slides (très brefs) & live demo
* Dispo ou indispo (après mars, dès que possible, pas entre X et Y): Session Journal Club de Mai, Multi-Sujets de Juin ?

**Comment by vfdev-5** [comment]
> Vous avez toujours rêvé de commencer le deep learning, ou une compétition Kaggle, mais n'avez pas de GPU chez vous ?

@fchouteau il y a déjà des GPUs sur kernel kaggle qu'on peut utiliser librement :)
